### PR TITLE
Allow consumers to override useInputValidation state, use in AddUserEmailForm

### PR DESCRIPTION
**Pull request by tjkandala**
@felixfbecker I think that while `useInputValidation`'s return value is starting to get ugly, most consumers don't need to override state. Should we prioritize giving consumers control, or is this change sufficient for our use cases?

### CODE REVIEW

File: client/shared/src/util/useInputValidation.ts [hunk 1]
```diff
@@ -66,7 +79,29 @@ export function useInputValidation(
...
+            // clear custom validity
+            inputReference.current?.setCustomValidity('')
+
+            // clear React state
+            setInputState({
+                kind: overrideOptions?.validate ? 'LOADING' : 'NOT_VALIDATED',
+                value: overrideOptions?.value ?? '',
+            })
+
+            if (overrideOptions?.value) {
```
Comment (felixfbecker) [inline client/shared/src/util/useInputValidation.ts:94]: Does this prevent overriding `{ value: '', validate: true }`?

File: client/web/src/user/settings/emails/AddUserEmailForm.tsx [hunk 2]
```diff
@@ -64,6 +64,7 @@ export const AddUserEmailForm: FunctionComponent<Props> = ({ user, className, on
             }
 
             eventLogger.log('NewUserEmailAddressAdded')
+            overrideEmailState()
```
Comment (felixfbecker) [inline client/web/src/user/settings/emails/AddUserEmailForm.tsx:67]: nit: I think requiring `{ value: '' }` to be passed would be a bit more readable.

**Review by felixfbecker — APPROVED** [review]
I think it's fine to merge this and you should own the experience and making further changes if you think it would be better (you could file an issue on yourself and include it in our iteration plan)

File: client/shared/src/util/useInputValidation.ts (hunk 1 above)
Comment (tjkandala) [reply client/shared/src/util/useInputValidation.ts:94]: Good catch, this should be `overrideOptions?.validate`, it doesn't matter what the value is

**Review by artemruts — APPROVED** [review]
Thank you, TJ!

**Pull request merged**